server/vulnerabilities/goval_dictionary/database.go

```diff
@@ -0,0 +1,71 @@
+package goval_dictionary
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/utils"
+	"strings"
+)
+
+type VulnDataSource interface {
+	// Eval evaluates the current OVAL definition against an OS version and a list of installed software, returns all software
+	// vulnerabilities found.
+	Eval(fleet.OSVersion, []fleet.Software) ([]fleet.SoftwareVulnerability, error)
+}
+
+func NewDB(db *sql.DB) *Database {
+	return &Database{sqlite: db}
+}
+
+type Database struct {
+	sqlite *sql.DB
+}
+
+// Eval evaluates the current goval_dictionary database against an OS version and a list of installed software,
+// returns all software vulnerabilities found.
+func (db Database) Eval(version fleet.OSVersion, software []fleet.Software) ([]fleet.SoftwareVulnerability, error) {
+	searchStmt := `SELECT packages.version, cves.cve_id 
```

Comment:
So, there's a tradeoff here.

We aren't specifying OS and version here, with the assumption that the sqlite files, as marked, will only ever include a single OS version. Given the naming convention of the files (`os_release`) this is safe, and allows us flexibility later to more aggressively thin the sqlite files (moving away from the full `goval-dictionary` output format to something that's much smaller), at the cost of needing to maintain one database per OS/version.

We can revisit this later, and based on file size comparisons I think this is the right choice (and this avoids the need to map our OS identifier to the getsion in `goval-dictionary`, which for AL2 would be "Amazon" for OS and "2" for version rather tahn `amzn_02`.

Reply:
This makes sense to me and continues the pattern of "only download the OVAL feeds you need"